R/derive_var_atoxgr.R

```diff
@@ -0,0 +1,287 @@
+#' Derive Lab toxicity grade 0 - 4
+#'
+#' @description
+#' Derives a character lab grade based on severity/toxicity criteria.
+#'
+#' @param dataset Input data set
+#'
+#'   The columns specified by `tox_description_var` parameter is expected.
+#'
+#' @param new_var Name of the character grade variable to create, for example, `ATOXGRH`
+#' or `ATOXGRL`.
+#'
+#' @param tox_description_var Variable containing the description of the grading
+#' criteria. For example: "Anemia" or "INR Increased".
+#'
+#' @param meta_criteria Metadata data set holding the criteria (normally a case statement)
+#'
+#'   Default: `atoxgr_criteria_ctcv4`
+#'
+#'   {admiral} metadata data set `atoxgr_criteria_ctcv4` implements
+#'   [Common Terminology Criteria for Adverse Events (CTCAE)
+#'    v4.0](https://ctep.cancer.gov/protocoldevelopment/electronic_applications/ctc.htm)
+#'
+#'   The metadata should have the following variables:
+#'
+#' - `TERM`: variable to hold the term describing the criteria applied to a particular lab test,
+#'   eg. "Anemia" or "INR Increased". Note: the variable is case insensitive.
+#' - `DIRECTION`: variable to hold the direction of the abnormality of a particular lab test
+#'   value. "L" is for LOW values, "H" is for HIGH values. Note: the variable is case insensitive.
+#' - `SI_UNIT_CHECK`: variable to hold unit of particular lab test. Used to check against input data
+#'   if criteria is based on absolute values.
+#' - `VAR_CHECK`: variable to hold comma separated list of variables used in criteria. Used to check
+#'   against input data that variables exist.
+#' - `GRADE_CRITERIA_CODE`: variable to hold code that creates grade based on defined criteria.
+#'
+#' @param criteria_direction Direction (L= Low, H = High) of toxicity grade.
+#' Permitted Values: "L", "H"
+#'
+#' @param get_unit_expr An expression providing the unit of the parameter
+#'
+#'   The result is used to check the units of the input parameters.
```

Comment:
Maybe clarify the extracted unit is checked against the `SI_UNIT_CHECK` from the metadata?

Reply:
Added more info.